guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc

```diff
@@ -56,25 +56,13 @@ Edit `/etc/hammer/cli_config.yml` and set the `:log_level:`:
 == Increasing the Logging Level On {SmartProxy}
 
 You can find the log for {SmartProxy} in `/var/log/foreman-proxy/proxy.log`.
-Uncomment the `DEBUG` line in `/etc/foreman-proxy/settings.yml`:
+Change the log level to `DEBUG`:
 
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-:log_level: DEBUG
+{foreman-installer} --foreman-proxy-log-level DEBUG
 ----
 
-Ensure to restart the `foreman-proxy` service afterwards:
-
-[options="nowrap", subs="+quotes,verbatim,attributes"]
-----
-# systemctl restart foreman-proxy
-----
-
-[CAUTION]
-====
-Running the installer will revert this change back.
-====
-
 ifdef::katello,orcharhino,satellite[]
 == Increasing the Logging Level For Candlepin
```

Comment:
I opened https://github.com/theforeman/puppet-candlepin/pull/226 to make this configurable via the installer.

Reply:
I now realize we can only change it view Hiera since there's no corresponding patch in puppet-katello.

Reply:
Looking at the changes in https://github.com/theforeman/puppet-candlepin/pull/226/, it seems that Candlepin logging can now also be configured using `foreman-installer`. Can you please confirm @ekohl?

Also, a vaguely related question: Is there a way to print all available log levels (debug, info, warn, ...) from CLI?

Reply:
> Looking at the changes in [theforeman/puppet-candlepin#226](https://github.com/theforeman/puppet-candlepin/pull/226), it seems that Candlepin logging can now also be configured using `foreman-installer`. Can you please confirm @ekohl?

While puppet-candlepin can do it, we don't expose that in the installer today. So you can configure it via custom-hiera.yaml, but we try not do document that since it's not intended for end users.

> Also, a vaguely related question: Is there a way to print all available log levels (debug, info, warn, ...) from CLI?

The [data type](https://github.com/theforeman/puppet-candlepin/blob/master/types/loglevel.pp) should validate the user input and if it fails, I think it should give a description but I don't think there's a way before to get the valid options. It would be a good feature request though.

Reply:
And I see I didn't link to https://github.com/theforeman/puppet-katello/pull/473 but that would expose it properly to end users.

The first instruction would be replaced with:
```
{foreman-installer} --katello-candlepin-loggers log4j.logger.org.candlepin:DEBUG
```

The second one:
```
{foreman-installer} \
--katello-candlepin-loggers log4j.logger.org.candlepin:DEBUG \
--katello-candlepin-loggers log4j.logger.org.candlepin.resource.ConsumerResource:WARN \
--katello-candlepin-loggers log4j.logger.org.candlepin.resource.HypervisorResource:WARN
```